lisa/sut_orchestrator/azure/features.py

```diff
@@ -1957,11 +1955,14 @@ def _call_requirement_method(self, method_name: str, capability: Any) -> Any:
             capability, ArchitectureSettings
         ), f"actual: {type(capability)}"
 
+        assert_that(self.arch).described_as(
+            "req and capability should be the same"
+        ).is_equal_to(capability.arch)
+
         value = ArchitectureSettings()
-        if self.arch or capability.arch:
-            value.arch = getattr(search_space, f"{method_name}_setspace_by_priority")(
-                self.arch, capability.arch, ["x64", "Arm64"]
-            )
+        value.arch = next(
```

Comment:
It doesn’t so complicated, just take one of them.

Reply:
updated